### PR TITLE
Add xml documentation inheritdoc conversion [#245]

### PR DESCRIPTION
**Pull request by robertlarkins**
This adds the inheritdoc tool to convert the inheritdoc tag in the xml documentation to the associated comments in System.IO. These comments are updated using those found in mscorlib.xml.

**Note:** The changes to appveyor.yml will need to be checked confirm they are operating correctly on appveyor.

### CODE REVIEW

**Comment by robertlarkins** [comment]
Will have to look at what caused the AppVeyor build error tomorrow.

**Comment by robertlarkins** [comment]
@fgreinacher This should hopefully be working now, but I don't have a way (that I know of) of checking that the xml documentation has been converted.

**Comment by fgreinacher** [comment]
Thanks for working on this @robertlarkins. For verification you could download the nupkgs from https://ci.appveyor.com/project/tathamoddie/system-io-abstractions/build/2.1.0.212-tcfasuhb/artifacts, extract them and then look at the “final” contents.

**Comment by robertlarkins** [comment]
@fgreinacher Just done _another_ fix and checked the artifact, the inheritdoc tags have now been replaced, so should be good to go.

File: System.IO.Abstractions/System.IO.Abstractions.csproj [hunk 1]
```diff
@@ -47,4 +47,10 @@
         <Reference Include="System" />
         <Reference Include="Microsoft.CSharp" />
     </ItemGroup>
+
+    <ItemGroup Condition="'$(TargetFramework)' == 'netstandard2.0'">
+      <PackageReference Include="InheritDoc">
```
[the remark this answers is not in the document]
Comment (robertlarkins) [reply System.IO.Abstractions/System.IO.Abstractions.csproj]: That's a good point, will look into it.

**Comment by robertlarkins** [comment]
InheritDoc isn't converting all the inheritdoc tags, so need to investigate why that is...

**Comment by ericnewton76** [comment]
This one looks mergeable without problems.

**Comment by robertlarkins** [comment]
@ericnewton76 don't merge yet, there's an issue with converting the inheritdoc tags to the actual comments that is being looked into. See here: https://github.com/firesharkstudios/InheritDoc/issues/3

**Comment by robertlarkins** [comment]
Wish I had found this half an hour ago: http://www.yamllint.com/

**Comment by robertlarkins** [comment]
@fgreinacher  @ericnewton76 InheritDoc appears to be fixed and the yaml build script is _finally_ working (after getting help on the appveyor forums). So take a look and see if it's good to go.

**Comment by fgreinacher** [comment]
Great - so here we go! Thanks for making this work!

**Pull request merged**